monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,153 @@ def get_params(self, image_size):
+
+        c, h, w = img_size
+
+        if img_size is not None:
+            pad_h = h % self.tile_size
+            pad_w = w % self.tile_size
+            if pad_h > 0 and pad_w > 0:
+                self.random_offset = (self.R.randint(pad_h), self.R.randint(pad_w))
+                h = h - self.random_offset[0]
+                w = w - self.random_offset[1]
+            else:
+                self.random_offset = None
+
+        if self.tile_pad_full:
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            h = h + pad_h
+            w = w + pad_w
+
+        h_n = (h - self.tile_size + self.tile_step) // self.tile_step
+        w_n = (w - self.tile_size + self.tile_step) // self.tile_step
+        tile_total = h_n * w_n
+
+        if tile_total > self.tile_count:
+            self.random_idxs = self.R.choice(range(tile_total), self.tile_count, replace=False)
+        else:
+            self.random_idxs = None
+
+    def __call__(self, image: np.ndarray) -> np.ndarray:
+
+        # add random offset
+        self.randomize(img_size=image.shape)
+
+        if self.tile_random_offset and self.random_offset is not None:
+            image = image[:, self.random_offset[0] :, self.random_offset[1] :]
+
+        # pad to full size, divisible by tile_size
+        if self.tile_pad_full:
+            c, h, w = image.shape
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            image = np.pad(
+                image,
+                [[0, 0], [pad_h // 2, pad_h - pad_h // 2], [pad_w // 2, pad_w - pad_w // 2]],
+                constant_values=self.tile_background_val,
+            )
+
+        # extact tiles (new way)
+        xstep, ystep = self.tile_step, self.tile_step
+        xsize, ysize = self.tile_size, self.tile_size
+        clen, xlen, ylen = image.shape
+        cstride, xstride, ystride = image.strides
+        llw = as_strided(
+            image,
+            shape=((xlen - xsize) // xstep + 1, (ylen - ysize) // ystep + 1, clen, xsize, ysize),
+            strides=(xstride * xstep, ystride * ystep, cstride, xstride, ystride),
+            writeable=False,
+        )
+        image = llw.reshape(-1, clen, xsize, ysize)
+
+        # if keep all patches
+        if self.tile_all:
+            # retain only patches with significant foreground content to speed up inference
+            # FYI, this returns a variable number of tiles, so the batch_size much be 1 (per gpu). Used during inference
+            thresh = 0.999 * 3 * self.tile_background_val * self.tile_size * self.tile_size
+            if self.tile_filter_mode == "min":
+                # default, keep non-background tiles (small values)
+                idxs = np.argwhere(image.sum(axis=(1, 2, 3)) < thresh)
```

Comment:
Can this transform work for both 2D and 3D data?

Thanks.

Reply:
No, it's currently for 2D only.  We don't have a reason to extend it to 3D, and scope of testing will be large

Reply:
Could you please help add some description in the doc-string of this transform for the input shape assumption? Because most other MONAI transforms support both 2D and 3D data.

Thanks.

Reply:
done